spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java

```diff
@@ -0,0 +1,198 @@
+import io.camunda.zeebe.spring.client.jobhandling.ZeebeClientExecutorService;
+import io.camunda.zeebe.spring.client.properties.ZeebeClientConfigurationProperties;
+import io.grpc.ClientInterceptor;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import jakarta.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static org.springframework.util.StringUtils.hasText;
+
+public class ZeebeClientConfiguration implements io.camunda.zeebe.client.ZeebeClientConfiguration {
+
+  @Autowired
+  private ZeebeClientConfigurationProperties properties;
+
+  @Autowired
+  private Authentication authentication;
+
+  @Lazy // Must be lazy, otherwise we get circular dependencies on beans
+  @Autowired
+  private JsonMapper jsonMapper;
+
+  @Lazy
+  @Autowired(required = false)
+  private List<ClientInterceptor> interceptors;
+
+  @Lazy
+  @Autowired
+  private ZeebeClientExecutorService zeebeClientExecutorService;
+
+  @PostConstruct
+  public void applyLegacy() {
+    // make sure environment variables and other legacy config options are taken into account (duplicate, also done by  qPostConstruct, whatever)
+    properties.applyOverrides();
+  }
+
+  @Override
+  public String getGatewayAddress() {
+    return properties.getGatewayAddress();
+  }
+
+  @Override
+  public String getDefaultTenantId() {
+    return properties.getDefaultTenantId();
+  }
+
+  @Override
+  public List<String> getDefaultJobWorkerTenantIds() {
+    return properties.getDefaultJobWorkerTenantIds();
+  }
+
+  @Override
+  public int getNumJobWorkerExecutionThreads() {
+    return properties.getNumJobWorkerExecutionThreads();
+  }
+
+  @Override
+  public int getDefaultJobWorkerMaxJobsActive() {
+    return properties.getDefaultJobWorkerMaxJobsActive();
+  }
+
+  @Override
+  public String getDefaultJobWorkerName() {
+    return properties.getDefaultJobWorkerName();
+  }
+
+  @Override
+  public Duration getDefaultJobTimeout() {
+    return properties.getDefaultJobTimeout();
+  }
+
+  @Override
+  public Duration getDefaultJobPollInterval() {
+    return properties.getDefaultJobPollInterval();
+  }
+
+  @Override
+  public Duration getDefaultMessageTimeToLive() {
+    return properties.getDefaultMessageTimeToLive();
+  }
+
+  @Override
+  public Duration getDefaultRequestTimeout() {
+    return properties.getDefaultRequestTimeout();
+  }
+
+  @Override
+  public boolean isPlaintextConnectionEnabled() {
+    return properties.isPlaintextConnectionEnabled();
+  }
+
+  @Override
+  public String getCaCertificatePath() {
+    return properties.getCaCertificatePath();
+  }
+
+  @Override
+  public CredentialsProvider getCredentialsProvider() {
+    if (!(authentication instanceof DefaultNoopAuthentication)) {
```

Comment:
I appreciate reverting the change. I'm happy to accept the PR, but would need to correct this condition so that users will not be stuck to always use override credentials headers in Zeebe. As it stands, there will be always Authentication object and it will not be `DefaultNoopAuthentication` provided they set the right properties.

My temporary plan / work-around would be to change the condition to check for flag `common.enabled` and then use this functionality. I'll make a release so we can have something usable for you but not breaking for others.

Expect this to be change substantially after that.

Thanks again!